src/main/java/com/nakal/ScreenExecutor/NakalExecutor.java

```diff
@@ -39,107 +37,85 @@ public NakalExecutor(){
         }
     }
 
-    private boolean buildMode(String baseLineImageName)
-        throws InterruptedException, IOException, IM4JavaException {
-        screenCaptureAndMaskRegionsIfPresent(baseLineImageName, baseLineImageName,
-            screenPaths.getExpectedImage(), screenPaths.getMaskedRegionExpectedImage(),
-            screenPaths.getMaskedExpectedImage());
-        return true;
+    private void captureScreen(String fileName, String imagepath){
+        new ScreenShooter().screenCapture(fileName,imagepath);
     }
 
-    private void initialize(String baseLineImageName) {
-        screenPaths.setExpectedImage(baseLineImageName);
-        screenPaths.setMaskedExpectedImage(baseLineImageName);
-        screenPaths.setMaskedRegionExpectedImage(baseLineImageName);
-        screenPaths.setMaskImage();
-        //Actual Params
-        screenPaths.setActualImage(baseLineImageName);
-        screenPaths.setActualMaskedRegionImage(baseLineImageName);
-        screenPaths.setDiffImage(baseLineImageName);
-        screenPaths.setMergedDiffImage(baseLineImageName);
-        screenPaths.setMaskedActualImage(baseLineImageName);
+    private void applyMaskRegion(String image,String maskedRegionImage,String baseLineImageName)
+            throws InterruptedException, IOException, IM4JavaException {
+        imageUtil.maskRegions(image, maskedRegionImage, baseLineImageName);
     }
 
-    public boolean nakalExecutorNativeCompare(String baseLineImageName, int threshold)
-        throws InterruptedException, IOException, IM4JavaException {
-        initialize(baseLineImageName);
-        if (isBuildMode()) {
-            return buildMode(baseLineImageName);
-        } else if (isCompareMode()) {
-            try {
-                screenCaptureAndMaskRegionsIfPresent(baseLineImageName,
-                     baseLineImageName, screenPaths.getActualImage(),
-                    screenPaths.getActualMaskedRegionImage(),
-                    screenPaths.getMaskedActualImage());
-
-                if (imageUtil.compareImages(screenPaths.getMaskedExpectedImage(),
-                        screenPaths.getMaskedActualImage(), screenPaths.getDiffImage(), threshold)) {
-                    file = new File(screenPaths.getDiffImage());
-                    file.delete();
-                    return true;
-                } else {
-                    mergerDiffHorizontal();
-                }
-
-            } catch (IOException e) {
-                e.printStackTrace();
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            } catch (IM4JavaException e) {
-                e.printStackTrace();
-            }
-        }
-        return Boolean.parseBoolean(null);
+    private void applyMaskImage(String maskedRegionImage, String maskedImage) throws InterruptedException, IOException, IM4JavaException {
+        imageUtil.maskImage(maskedRegionImage, screenPaths.getMaskImage(),
```

Comment:
difference between applyMaskImage and applyMaskRegion? I think we need to rename this? what say.